src/partr.c

```diff
@@ -389,12 +389,14 @@ JL_DLLEXPORT jl_task_t *jl_task_get_next(jl_value_t *trypoptask, jl_value_t *q,
 {
     jl_task_t *ct = jl_current_task;
     uint64_t start_cycles = 0;
-
+    uint64_t t0 = jl_hrtime();
     while (1) {
+        jl_ptls_t ptls = ct->ptls;
         jl_task_t *task = get_next_task(trypoptask, q);
-        if (task)
+        if (task) {
+            ptls->timing_tls.scheduler_time += jl_hrtime() - t0;
             return task;
-
+        }
```

Comment:
this is the fast-path for task-switches (i think?), but i think we've concluded this shouldn't add too much overhead (given `jl_hrtime` is a `vdso` call)... still need to verify that experimentally though